scripts/world_bank/datasets/datasets.py

```diff
@@ -0,0 +1,240 @@
+VERSION_ID_PATTERN = '?versionId='
+
+
+def to_dataset_csv_row(resource, dataset_name, view_file):
+    format = resource.get('format')
+    if format is None:
+        return None
+
+    url = None
+    for field in DATASET_URL_FIELDS:
+        url = resource.get(field)
+        if url is not None:
+            url = url.split(VERSION_ID_PATTERN)[0]
+            break
+
+    if url is None:
+        return None
+
+    return {
+        DATASET_NAME_COLUMN_NAME: dataset_name,
+        DATASET_RESOURCE_NAME_COLUMN_NAME: resource.get('name'),
+        DATASET_VIEW_FILE_COLUMN_NAME: view_file,
+        DATASET_FORMAT_COLUMN_NAME: format,
+        DATASET_SIZE_COLUMN_NAME: resource.get('distribution_size'),
+        DATASET_DOWNLOAD_URL_COLUMN_NAME: url
+    }
+
+
+def fetch_dataset_view(dataset_unique_id):
+    params = f"dataset_unique_id={dataset_unique_id}"
+    response_file = f"{DATASET_VIEWS_RESPONSE_DIR}/view-{dataset_unique_id}.json"
+    return load_json(url=DATASET_VIEW_URL,
+                     params=params,
+                     response_file=response_file)
+
+
+def fetch_dataset_views():
+    # Assumes all dataset list responses have been saved to disk
+    list_files = os.listdir(DATASET_LISTS_RESPONSE_DIR)
+    dataset_unique_ids = []
+    for list_file in list_files:
+        json = load_json_file(f"{DATASET_LISTS_RESPONSE_DIR}/{list_file}")
+        for dataset in json.get('data', []):
+            dataset_unique_id = dataset.get('dataset_unique_id')
+            if dataset_unique_id:
+                dataset_unique_ids.append(dataset_unique_id)
+
+    with multiprocessing.Pool(POOL_SIZE) as pool:
+        pool.starmap(fetch_dataset_view, zip(dataset_unique_ids))
+
+
+def fetch_dataset_list(skip):
+    params = f"$skip={skip}&$top=1000"
+    response_file = f"{DATASET_LISTS_RESPONSE_DIR}/list-{skip}.json"
+    return load_json(url=DATASET_LIST_URL,
+                     params=params,
+                     response_file=response_file)
+
+
+def fetch_dataset_lists():
+    # 6000+ datasets, fetching 1000 at a time
+    skips = range(0, 7000, 1000)
+    with multiprocessing.Pool(POOL_SIZE) as pool:
+        pool.starmap(fetch_dataset_list, zip(skips))
+
+
+def load_json(url, params, response_file):
+    if os.path.exists(response_file):
+        logging.info('Reading response from file %s', response_file)
+        with open(response_file, 'r') as f:
+            return json.load(f)
+
+    logging.info("Fetching url %s, params %s", url, params)
+    response = requests.get(url, params=params).json()
+    with open(response_file, 'w') as f:
+        logging.info('Writing response to file %s', response_file)
+        json.dump(response, f, indent=2)
+    return response
+
+
+def load_json_file(json_file):
+    with open(json_file, 'r') as f:
+        return json.load(f)
+
+
+def main(_):
+    fetch_and_write_datasets_csv()
+    # download_datasets()
```

Comment:
can we add a command line flag for this so user won't have to edit the script to download datasets?

Example:
https://docs.google.com/document/d/1asu3CdLHd4xZZIb7oLmVdiaMOO3EQl7BYvD6c0d41pU/edit#heading=h.6mh8op9lb42r

Reply:
Done.